From a1993266d00dc941f0494d2f825c276b8e05b5cb Mon Sep 17 00:00:00 2001
From: Hongcheng Xie <hongcheng.xie@intel.com>
Date: Thu, 7 Apr 2022 11:20:49 +0800
Subject: [PATCH] Don't restart system_server even lost network

Don't throw fatal exception for android cloud scenario even
lost network stack error happen, the network stack may restore
in some case such as networkstack process crash and restart.

Tracked-On: OAM-101469
Signed-off-by: Hongcheng Xie <hongcheng.xie@intel.com>
Change-Id: Id71cc2b42632eb97334c46a4e74765479d8ddbc3
---
 .../net/java/android/net/ConnectivityModuleConnector.java | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/services/net/java/android/net/ConnectivityModuleConnector.java b/services/net/java/android/net/ConnectivityModuleConnector.java
index 62f2c35f339b..2012cab2938c 100644
--- a/services/net/java/android/net/ConnectivityModuleConnector.java
+++ b/services/net/java/android/net/ConnectivityModuleConnector.java
@@ -31,6 +31,7 @@ import android.os.Environment;
 import android.os.IBinder;
 import android.os.Process;
 import android.os.SystemClock;
+import android.os.SystemProperties;
 import android.os.UserHandle;
 import android.provider.DeviceConfig;
 import android.text.format.DateUtils;
@@ -332,7 +333,12 @@ public class ConnectivityModuleConnector {
             // Using device-encrypted SharedPreferences as DeviceConfig does not have a synchronous
             // API to persist settings before a crash.
             tryWriteLastCrashTime(prefs, now);
-            throw new IllegalStateException(message);
+            // Don't throw fatal exception for android cloud scenario even lost network stack
+            if (SystemProperties.getBoolean("ro.enable.android.cloud", true)) {
+                logWtf("Don't throw fatal exception even lost network in aic", null);
+            } else {
+                throw new IllegalStateException(message);
+            }
         }
 
         // Here the system crashed recently already. Inform listeners that something is
-- 
2.25.1

